.github/workflows/cf-driver-go.yml

```diff
@@ -0,0 +1,37 @@
+# This workflow will build a golang project
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go
+
+name: "CF Driver: Go Build & Test"
+
+on: [pull_request]
+
+defaults:
+  run:
+    working-directory: runner/cf-driver-go
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
```

Comment:
[question] can we start out with `v5` of the action?

Reply:
Probably? I didn't realized I'd picked an old one, can you tell me how this came up for you?

Reply:
I checked the action readme to see if we could avoid passing `go-version` (seems like no? but if we can just have it inherit from `go.mod` that'd be ideal) and saw that v5 was out

Reply:
Ah, I was hoping some automated thing had surfaced it :-/

But yeah, that's what I had initially planned to do, but then something I read in documentation made me think specifying a version would result in GitHub being more likely to pull a cached version of Go--and that seemed like a nice enough upside to me. I mean to verify it actually made a difference, though, and I forgot to do that.

Reply:
So, it _does_ make a difference, but not a particularly large difference. Specifying a `1.23.x` version for the action takes < 1s, using the `go.mod` takes ~8s—_after_ the initial run which was an extra ~22s, but I'm thinking this probably wouldn't happen often.

I think for 8s it probably isn't worth it, and it might be faster if I don't specify a patch version in `go.mod`, which we have no reason to do.